beanquery/query_env.py

```diff
@@ -571,6 +572,34 @@ def date_add(x, y):
     return x + datetime.timedelta(days=y)
 
 
+@function([str, datetime.date], datetime.date)
+def date_trunc(field, x):
+    """Truncates the specified field of a given date."""
+    if field == "month":
+        return datetime.date(x.year, x.month, 1)
+    elif field == "year":
+        return datetime.date(x.year, 1, 1)
+    else:
+        return None
+
+
+@function([str], relativedelta)
+def interval(x):
+    """Returns a relative time interval."""
+    parts = x.replace("- ", "-").split()
```

Comment:
This handles intervals joined with the `-` operator, but not intervals joined with the `+` operator. I think both should be handled, or neither. I think it also breaks for intervals like `-1 days`. Finally, it does not work correctly for intervals like `1 day + 2 days`, which I believe is handled by other SQL implementations.

I would keep it easy and just handle intervals in the form `[0-9]+ {specifier}`.

Reply:
I implemented the `+`, specifying the same specified multiple times and added additional tests.

Main reason why I'd like this is so I can write
```
SELECT date_trunc("month", date) + interval("1 months - 1 days")
```
instead of the more verbose form
```
SELECT date_trunc("month", date) + interval("1 months") - interval("1 days")
```

Reply:
Agree, I've dropped the support for complex expressions now. Maybe I or someone else can look into that in a future PR.